core/nuxeo-operation.js

```diff
@@ -199,21 +199,54 @@ import './nuxeo-connection.js';
     execute() {
       this._setActiveRequests(this.activeRequests + 1);
 
-      const params = !this.params || typeof this.params === 'object' ? this.params : JSON.parse(this.params);
+      let params = !this.params || typeof this.params === 'object' ? this.params : JSON.parse(this.params);
 
       let { input } = this;
 
-      // support page provider as input to operations
-      // relies on parameters naming convention until provider marshaller is available
-      if (Nuxeo.PageProvider && input instanceof Nuxeo.PageProvider) {
-        params.providerName = input.provider;
-        Object.assign(params, input._params);
+      let { op } = this;
+
+      // the goal is to track if we have a bulk action in select all mode (to be used for the abort method)
+      let isBulk = false;
+
+      if (this._isPageProvider(input) || this._isView(input)) {
+        let pageProvider;
+        // support page provider as input to operations
+        // relies on parameters naming convention until provider marshaller is available
+        if (this._isPageProvider(input)) {
+          pageProvider = input;
+          input = undefined;
+        } else if (this._isSelectAllActive(input)) {
+          // in select all mode, we use `Bulk.RunAction` as the operation and `this.op` as a parameter for it
+          op = 'Bulk.RunAction';
+          // support page provider display behavior instances (table, grid, list) as input to operations for select all
+          pageProvider = input.nxProvider;
+          params = {
+            action: 'automationUi',
+            providerName: pageProvider.provider,
+            parameters: JSON.stringify({
+              operationId: this.op,
+              parameters: params,
+            }),
+          };
+          isBulk = true;
+          input = undefined;
+        } else {
+          // only a few selected documents should be considered for the operation
+          pageProvider = input.nxProvider;
+          const uids = input.selectedItems.map((doc) => doc.uid);
+          const uidsString = uids.join(',');
+          input = `docs:${uidsString}`;
```

Comment:
I think that this could just be:
```suggestion
          input = input.selectedItems;
```

Reply:
Please test this @richardsd ⬆️

Reply:
I will 👍

Reply:
It works 🎉 From what I see in my tests (and the http requests in the network panel) the string `docs:${uids}` is being passed to the rest API, which means that the JS client is actually handling this 🎉